### PR TITLE
TCK Migration: Move jaxrs/spec tests from jakartaee-tck

### PR DESCRIPTION
**Pull request by alwin-joseph**
This change is in continuation of TCK migration from https://github.com/eclipse-ee4j/jaxrs-api/pull/1002 .

 To test the changes :

export JAVA_HOME=<JDK11_HOME>
export M2_HOME=<MAVEN_HOME>
export PATH=$M2_HOME/bin:$JAVA_HOME/bin:$PATH

jaxrs-tck/
mvn clean install

jersey-tck/
run all tests : `mvn clean verify -Parq-glassfish-managed`
skip optional tests : `mvn clean verify -Parq-glassfish-managed -DexcludedGroups="xml_binding"`  

The tests migrated :
jakartaee-tck/src/com/sun/ts/tests/jaxrs/spec(317)

Current status : 
1 test failure in jakarta.ws.rs.tck.spec.provider.overridestandard.JAXRSClientIT.readWriteStreamingOutputProviderTest is commented for now. This will be tracked at https://github.com/eclipse-ee4j/jaxrs-api/issues/1020

2 tests are disabled as it was in jakartaee-tck:
jakarta.ws.rs.tck.spec.resource.responsemediatype.JAXRSClientIT.noProducesTest
jakarta.ws.rs.tck.spec.resource.responsemediatype.JAXRSClientIT.noProducesResponseReturnTest

### CODE REVIEW

**Comment by alwin-joseph** [comment]
@mkarg @andymc12 @spericas @gurunrao

**Comment by alwin-joseph** [comment]
> `jakarta.ws.rs.tck.spec.provider.standardhaspriority.JAXRSClientIT.readWriteJaxbProviderTest` is a JAX-B test, but JAX-B is optional in 3.1.
> Same with `jakarta.ws.rs.tck.spec.filter.interceptor.JAXRSClientIT.jaxbWriterClientInterceptorTest`,
> `jakarta.ws.rs.tck.spec.client.typedentitieswithxmlbinding.JAXRSClientIT.clientJaxbElementReaderTest`, `jakarta.ws.rs.tck.spec.client.typedentitieswithxmlbinding.JAXRSClientIT.clientJaxbElementWriterTest`, and `jakarta.ws.rs.tck.spec.provider.standardhaspriority.JAXRSClientIT.readWriteJaxbProviderTest`.

Tagged the above 4 tests as optional.
The optional tests can be excluded by running `mvn clean verify -Parq-glassfish-managed -DexcludedGroups="optional"`

**Review by gurunrao — COMMENTED** [review]
is it possible to squash commits for better readability of repository history?

**Comment by gurunrao** [comment]
run all tests logs : [run.log](https://github.com/eclipse-ee4j/jaxrs-api/files/7241202/run.log)

skip optional tests logs: [skip_optional_run.log](https://github.com/eclipse-ee4j/jaxrs-api/files/7241206/skip_optional_run.log)

File: jaxrs-tck/src/main/java/jakarta/ws/rs/tck/spec/provider/standard/JAXRSClientIT.java [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+   * JAXRS:SPEC:36
+   * 
+   * @test_Strategy: MessageBodyReader providers always operate on the decoded
+   * HTTP entity body rather than directly on the HTTP message body.
+   * 
+   * This is not possible with JAXRS Client.
+   * 
+   * This depends on a container, works in TC, not in GF
+   * 
+   * public void encodedEntityTest() throws Fault { for (String method :
```
[the remark this answers is not in the document]
Comment (alwin-joseph) [reply jaxrs-tck/src/main/java/jakarta/ws/rs/tck/spec/provider/standard/JAXRSClientIT.java:310]: This tests is commented in jakartaee-tck hence it was retained it that way. Is there any reason to enable it and run ?

[the remark this answers is not in the document]
Comment (gurunrao) [reply jaxrs-tck/src/main/java/jakarta/ws/rs/tck/spec/provider/standard/JAXRSClientIT.java:310]: > This tests is commented in jakartaee-tck hence it was retained it that way. Is there any reason to enable it ?

We can keep the test commented post migration too, if test commented out in jakartaee-tck.

File: jaxrs-tck/src/main/java/jakarta/ws/rs/tck/spec/provider/jaxbcontext/JAXRSClientIT.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+   * 
+   * @assertion_ids: JAXRS:SPEC:34
+   * 
+   * @test_Strategy: The implementation-supplied entity provider(s) for
+   * jakarta.xml.bind.JAXBElement and application supplied JAXB classes MUST use
+   * JAXBContext instances provided by application-supplied context resolvers,
+   * see Section 4.3.
+   */
+  @Test
+  public void readWriteProviderTest() throws Fault {
```
[the remark this answers is not in the document]
Comment (alwin-joseph) [reply jaxrs-tck/src/main/java/jakarta/ws/rs/tck/spec/provider/jaxbcontext/JAXRSClientIT.java:99]: marked the xml_binding tests as per jakartaee-tck https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/src/com/sun/ts/lib/harness/keyword.properties

**Comment by alwin-joseph** [comment]
> is it possible to squash commits for better readability of repository history?

The PR  will be squashed and merged post review as it was done for previous PR with test migration.

**Review by jansupol — DISMISSED** [review]

**Review by gurunrao — APPROVED** [review]
LGTM

**Review by spericas — APPROVED** [review]
LGTM. But I would like that at some point that we re-format this code to follow the Jakarta REST guidelines for indentation, etc.

**Review by jansupol — APPROVED** [review]

**Review by gurunrao — APPROVED** [review]

**Review by m0mus — APPROVED** [review]
LGTM

**Comment by jansupol** [comment]
I am pretty sure I tried to squash this, but it did not happen...

**Pull request merged**